### PR TITLE
Scala3doc: Fix inproper rendering of HTML documentation elements.

### PR DESCRIPTION
**Pull request by pikinier20**

### CODE REVIEW

File: scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala [hunk 1]
```diff
@@ -40,8 +40,13 @@ trait SyntheticsSupport:
   def isSyntheticField(c: Symbol) =
     c.flags.is(Flags.CaseAccessor) || (c.flags.is(Flags.Module) && !c.flags.is(Flags.Given))
 
+  // There seems to be no straightforward way to check if position exists so we catch AssertionError defaulting to true
```
Comment (abgruszecki) [inline scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala:43]: We should introduce a `hack*` method that checks if a `Position` exists. The fact that a `Position` can not exist is a bug in the Reflect API, there's an issue for it.

File: scala3doc/src/dotty/renderers/MemberRenderer.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ import collection.JavaConverters._
 
 class MemberRenderer(signatureRenderer: SignatureRenderer, buildNode: ContentNode => String)(using DocContext):
 
-  private val converter = new DocTagToContentConverter()
+  private val converter = ScalaCommentToContentConverter
```
Comment (abgruszecki) [inline scala3doc/src/dotty/renderers/MemberRenderer.scala:13]: Shouldn't this be injected?

Comment (romanowski) [reply scala3doc/src/dotty/renderers/MemberRenderer.scala:13]: This is a code that intend to not dependend on dokka as much as possible hence no injection

File: scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala (hunk 1 above)
Comment (pikinier20) [reply scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala:45]: Done

File: scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala [hunk 3]
```diff
@@ -37,14 +37,13 @@ trait SyntheticsSupport:
 
     def getAllMembers: List[Symbol] = hackGetAllMembers(using qctx)(s)
 
+    def hasValidPos = Option.when(hackHasPosition(using qctx)(s))(s).flatMap(_.pos).fold(true)(p => p.start != p.end)
```
Comment (abgruszecki) [inline scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala:40]: Shouldn't this default to `false` if option is empty?

**Review by romanowski — APPROVED** [review]

File: scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala (hunk 3 above)
Comment (pikinier20) [reply scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala]: Yes, I've changed it to false.

**Comment by abgruszecki** [comment]
@romanowski I'm watching the CI on this one, I'll merge it if the community build takes too long.

**Pull request merged**